zoom_rooms/indico_zoom_rooms/util.py

```diff
@@ -0,0 +1,94 @@
+# This file is part of the CERN Indico plugins.
+# Copyright (C) 2024 CERN
+#
+# The CERN Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License; see
+# the LICENSE file for more details.
+
+import itertools
+import typing as t
+
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.models.events import Event
+from indico.modules.events.sessions.models.blocks import SessionBlock
+from indico.modules.events.sessions.models.sessions import Session
+from indico.modules.rb.models.room_attributes import RoomAttribute
+from indico.modules.rb.models.rooms import Room
+from indico.modules.vc.models.vc_rooms import VCRoom, VCRoomEventAssociation
+
+
+def get_zoom_room_id(obj: Event | Contribution | SessionBlock | Room) -> str | None:
+    """Get the Zoom Room Exchange user id for an object (e.g. event) or room.
+
+    Returns `None` if there isn't one.
+    """
+    room = obj if isinstance(obj, Room) else obj.room
+
+    if (
+        room
+        and (attributes := room.attributes)
+        and (
+            user_id_attr := attributes.filter(RoomAttribute.name == 'zoom-rooms-calendar-id')
+            .join(RoomAttribute)
+            .first()
+        )
+    ):
```

Comment:
really? :D 

(not a fan of running queries inside an `if` unless they all fit in a single line)

Reply:
I wanted to avoid nesting ifs. TBH I find them equally readable. But I can change it, no strong opinion.

Reply:
I just realized that `room.attributes` is either never falsy or triggers a query when used in a bool context. So not a great idea in either case.

But I think you can replace this *whole* construct with this:

```py
if room:
    return room.get_attribute_value('zoom-rooms-calendar-id')
```

Reply:
Perfect. Thanks!